components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/common/InboundRunner.java

```diff
@@ -71,8 +75,13 @@ public void run() {
         while (!init) {
             log.debug("Waiting for the configuration context to be loaded to run Inbound Endpoint.");
             Boolean isSinglNode = ClusteringAgentUtil.isSingleNode();
+            Parameter clusteringPattern = ServiceReferenceHolder.getInstance().getConfigurationContextService().
+                    getServerConfigContext().getAxisConfiguration().getClusteringAgent().
+                    getParameter(CLUSTERING_PATTERN);
+            Boolean isWorkerManager = clusteringPattern != null && clusteringPattern.getValue() != null
+                    && clusteringPattern.getValue().toString().equals(CLUSTERING_PATTERN_WORKER_MANAGER);
             if (isSinglNode != null) {
-                if (!isSinglNode && !CarbonUtils.isWorkerNode() && !runOnManagerOverride) {
+                if (!isSinglNode && !CarbonUtils.isWorkerNode() && !runOnManagerOverride && isWorkerManager) {
```

Comment:
Is there any possibility this while loop run infinitely? Do we have considered exit condition for this loop for all cases?